plugins/eventmonitor/eventmonitor.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+  styleinspector.cpp
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2012-2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Volker Krause <volker.krause@kdab.com>
+
+  Licensees holding valid commercial KDAB GammaRay licenses may use this file in
+  accordance with GammaRay Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "eventmonitor.h"
+
+#include "eventmodelroles.h"
+
+#include <core/aggregatedpropertymodel.h>
+#include <core/metaobjectrepository.h>
+#include <core/metaobject.h>
+#include <core/objectinstance.h>
+#include <core/remote/serverproxymodel.h>
+
+#include <common/objectbroker.h>
+#include <common/objectmodel.h>
+
+#include <QDebug>
+#include <QItemSelectionModel>
+#include <QMutex>
+#include <QSortFilterProxyModel>
+
+using namespace GammaRay;
+
+
+static EventModel *s_model = nullptr;
+
+
+QString eventTypeToClassName(QEvent::Type type) {
+    switch (type) {
+    case QEvent::NonClientAreaMouseMove:
+    case QEvent::NonClientAreaMouseButtonPress:
+    case QEvent::NonClientAreaMouseButtonRelease:
+    case QEvent::NonClientAreaMouseButtonDblClick:
+    case QEvent::MouseButtonDblClick:
+    case QEvent::MouseButtonPress:
+    case QEvent::MouseButtonRelease:
+    case QEvent::MouseMove:
+        return "QMouseEvent";
```

Comment:
This would benefit from QStringLiteral I think, as this is potentially a hot path (called for every event) and that would avoid memory allocation and text codec conversions here.